packages/typespec-python/test/azure/generated/azure-mgmt-spheredpg/azure/mgmt/spheredpg/models/_models.py

```diff
@@ -142,8 +144,28 @@ def __init__(self, mapping: Mapping[str, Any]):
         """
 
     def __init__(self, *args: Any, **kwargs: Any) -> None:  # pylint: disable=useless-super-delegation
+        flatten_map = {k: kwargs.pop(k, None) for k in self._flatten_items}
         super().__init__(*args, **kwargs)
-
+        for k, v in flatten_map.items():
+            if v is not None:
+                setattr(self, k, v)
+
+    def __getattr__(self, name: str) -> Any:
+        if name in self._flatten_items:
```

Comment:
we can leave the `raise AttributeError` to the parent class by passing the property to super, so it can look more like this
```python
def __getattr__(self, name: str) -> Any:
  if name in self._flatten_items:
    return getattr(self.properties, name) if self.properties else `None`
  return super().__getattr__(name)
```

Reply:
https://github.com/Azure/autorest.python/pull/2251#discussion_r1406953092